topics/wallet-interaction-standard.md

```diff
@@ -0,0 +1,257 @@
+
+### `permission`
+
+The purpose of the `permission` messages is to establish a connection between a relying party and a wallet, grant the relying party access to public parts of the user's identity and define the scope of actions the relying part is allowed to perform.
+
+#### Types
+
+- `text`: A plain `string` value.
+- `blob`: A `string` value describing binary data encoded in base64.
+
+#### Request
+
+`version` (`text`): The version of the standard used. If the wallet does not support the version of the request, it must send the `"VERSION_NOT_SUPPORTED"` error in response.
+
+`appMetadata`: Information about the relying party.
+- `name` (`text`): The user-friendly name of the relying party. 
+- `icon` (`text`, optional): An optional URL pointing to an icon resource representing the relying party.
+
+`networks`: A list of networks on which the relying party plans to operate.
+- `chainId`(`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes the relying party requires. If the wallet does not support any of the requested scopes, it should ignore it. Possible values: 
+- `"canister_call"`
+
+`challenge` (`blob`): A challenge used for the wallet to sign in order to prove its access to the identity. The challenge should be an array of 32 cryptographically random bytes generated from a secure random source by the sender of the request.
+
+#### Response
+
+`version` (`text`): The version of the standard used. It must match the `version` from the request.
+
+`appMetadata`: Information about the wallet.
+- `name` (`text`): The user-friendly name of the wallet.
+- `icon` (`text`, optional): An optional URL poiting to an icon resource representing the wallet.
+
+`networks`: A list of networks on which the user has agreed the relying party can operate. This should be a subset of the `networks` from the original request.
+- `chainId` (`text`): The chain id of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes that the wallet supports and the user has agreed the relying party can be granted. This should be a subset of the `scopes` field from the original request. Possible values:
+- `"canister_call"`
+
+`identities`: A list of identities the user has selected.
+- `publicKey` (`blob`): The DER-encoded public key associated with the identity. The public key can be used to [derive a self-authenticating principal](https://internetcomputer.org/docs/current/references/ic-interface-spec/#principal).
+- `ledger` (optional): Ledger data.
+    - `subaccounts` (`blob`, optional): A list of subaccounts the user has selected.
```

Comment:
Could we add an optional `name` for the entries of `subaccounts`?

Reply:
@frederikrothenberger Where would we put the blob when subaccounts becomes and object? I currently implemented this structure:
```ts
subAccount: {
    bytes: blob;
    name?: string;
}
```

Reply:
Yes, but it's an array of objects, so:
```
subAccounts: {
    bytes: blob;
    name?: string;
}[]
```

Reply:
I think that is consistent with how accounts are defined in [ICRC-1](https://github.com/dfinity/ICRC-1/blob/main/standards/ICRC-1/README.md):
`type Account = record { owner : principal; subaccount : opt Subaccount; };`
I.e. you can leave it off completely.